WordPress/src/main/java/org/wordpress/android/ui/stockmedia/StockMediaPickerActivity.java

```diff
@@ -0,0 +1,590 @@
+        }
+    }
+
+    private void uploadSelection() {
+        ToastUtils.showToast(this, "Uploading will be added in a later PR");
+        // List<StockMediaModel> items = mAdapter.getSelectedStockMedia();
+    }
+
+    class StockMediaAdapter extends RecyclerView.Adapter<StockViewHolder> {
+        private static final float SCALE_NORMAL = 1.0f;
+        private static final float SCALE_SELECTED = .8f;
+
+        private final List<StockMediaModel> mItems = new ArrayList<>();
+        private final ArrayList<Integer> mSelectedItems = new ArrayList<>();
+
+        StockMediaAdapter() {
+            setHasStableIds(true);
+        }
+
+        void setMediaList(@NonNull List<StockMediaModel> mediaList) {
+            mItems.clear();
+            mItems.addAll(mediaList);
+            notifyDataSetChanged();
+        }
+
+        void addMediaList(@NonNull List<StockMediaModel> mediaList) {
+            mItems.addAll(mediaList);
+            notifyDataSetChanged();
+        }
+
+        void clear() {
+            mItems.clear();
+            if (mSelectedItems.size() > 0) {
+                mSelectedItems.clear();
+                notifySelectionCountChanged();
+            }
+            notifyDataSetChanged();
+        }
+
+        @Override
+        public long getItemId(int position) {
+            return mItems.get(position).getId().hashCode();
+        }
+
+        @Override
+        public int getItemCount() {
+            return mItems.size();
+        }
+
+        boolean isEmpty() {
+            return getItemCount() == 0;
+        }
+
+        @Override
+        public StockViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+            View view = getLayoutInflater().inflate(R.layout.stock_media_picker_thumbnail, parent, false);
+            return new StockViewHolder(view);
+        }
+
+        @Override
+        public void onBindViewHolder(StockViewHolder holder, int position) {
+            StockMediaModel media = mItems.get(position);
+            String imageUrl = PhotonUtils.getPhotonImageUrl(media.getThumbnail(), mThumbWidth, mThumbHeight);
+            holder.mImageView.setImageUrl(imageUrl, WPNetworkImageView.ImageType.PHOTO);
+
+            boolean isSelected = isItemSelected(position);
+            holder.mSelectionCountTextView.setSelected(isSelected);
+            if (isSelected) {
+                int count = mSelectedItems.indexOf(position) + 1;
+                String label = Integer.toString(count);
+                holder.mSelectionCountTextView.setText(label);
+            } else {
+                holder.mSelectionCountTextView.setText(null);
+            }
+
+            float scale = isSelected ? SCALE_SELECTED : SCALE_NORMAL;
+            if (holder.mImageView.getScaleX() != scale) {
+                holder.mImageView.setScaleX(scale);
+                holder.mImageView.setScaleY(scale);
+            }
+
+            if (!mIsFetching && mCanLoadMore && position == getItemCount() - 1) {
```

Comment:
I think it'd be much cleaner to have the `!mIsFetching && mCanLoadMore` checks in the `fetchStockMedia`. That way the adapter doesn't really care about it and when another developer looks at the `fetchStockMedia` method, they can tell when a new request will happen and when it won't. It makes it easier to catch any unexplored paths as well. What do you think?

Reply:
I agree about checking `mIsFetching` in `fetchStockMedia` but I think `mCanLoadMore` should continue to be checked in the adapter. I'll make this change shortly.